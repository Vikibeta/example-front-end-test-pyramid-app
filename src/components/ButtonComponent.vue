<template>
  <button
    @click="toggleModal"
    class="button"
  >
    Open modal
  </button>
</template>

<script>
import { mapActions } from 'vuex'

export default {
  name: 'button-component',
  methods: mapActions([
    'toggleModal'
  ])
}
</script>
