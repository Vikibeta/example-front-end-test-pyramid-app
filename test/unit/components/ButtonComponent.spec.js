import { createLocalVue, shallow } from 'vue-test-utils'
import { createRenderer } from 'vue-server-renderer'
import Vuex from 'vuex'
import ButtonComponent from '../../../src/components/ButtonComponent'

describe('ButtonComponent.vue', () => {
  test('calls toggleModal when button is clicked', () => {
    const actions = {
      toggleModal: jest.fn()
    }
    const localVue = createLocalVue()
    localVue.use(Vuex)
    const store = new Vuex.Store({
      actions
    })
    const wrapper = shallow(ButtonComponent, {
      localVue,
      store
    })
    wrapper.find('button').trigger('click')
    expect(actions.toggleModal).toHaveBeenCalled()
  })

  it('has same HTML structure', () => {
  const localVue = createLocalVue()
  localVue.use(Vuex)
  const store = new Vuex.Store({
    actions: {toggleModal: () => {}},
  })
  const renderer = createRenderer()
  const wrapper = shallow(ButtonComponent, {
    localVue,
    store
  })

  renderer.renderToString(wrapper.vm, (err, str) => {
    if (err) throw new Error(err)
    expect(str).toMatchSnapshot()
  })
})
})
