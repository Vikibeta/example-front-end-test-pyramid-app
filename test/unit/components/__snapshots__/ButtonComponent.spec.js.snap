// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonComponent.vue has same HTML structure 1`] = `
<button class="button">
  Open modal
</button>
`;
